### PR TITLE
feature: main&notifications/add properties list

### PR DESCRIPTION
**Pull request by rudenkoserhii**
## Describe your changes

- Add properties list component with fetching from CRM.

## Issue ticket code (and/or) and link

- [[Link to JIRA ticket](https://testhomework4.atlassian.net/browse/AG-126?atlOrigin=eyJpIjoiNTdmN2ZhYjM1MTU3NDdjMzg5M2Q0MmVkOTJiYTRjMDQiLCJwIjoiaiJ9)]

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [ ] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [ ] No inline styles
- [ ] Imports are absolute
- [ ] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/api/property.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { useMemo } from 'react';
+import useSWR from 'swr';
+// utils
```
Comment (canterbery) [inline src/api/property.ts]: unnecessary

File: src/types/property.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
+// ----------------------------------------------------------------------
```
Comment (canterbery) [inline src/types/property.ts]: same

File: src/utils/axios.ts [hunk 3]
```diff
@@ -1,25 +1,41 @@
...
 // ----------------------------------------------------------------------
+const axiosInstance = axios.create({ baseURL: BASE_URL });
 
-const axiosInstance = axios.create({ baseURL: HOST_API });
+axiosInstance.interceptors.response.use(
+  (res) => res,
+  (error) => Promise.reject((error.response && error.response.data) || 'Something went wrong')
+);
+
+// ----------------------------------------------------------------------
```
Comment (canterbery) [inline src/utils/axios.ts]: same

Comment (rudenkoserhii) [reply src/utils/axios.ts]: made as common structure of the template

Comment (canterbery) [reply src/utils/axios.ts]: Yes, but we were told to remove those anyway.  Also you development check failed - you should make a lint check of your branch

File: src/app/[locale]/main-page/page.tsx [hunk 4]
```diff
@@ -1,10 +1,11 @@
 'use client';
 
+import PropertiesList from 'components/custom/propertiesList';
 import { Box, Typography } from '@mui/material';
 import { ProtectedRoute } from 'components/custom';
 import { NotificationCenter, PropertyFilter } from './lib';
```
Comment (AKutusenko) [inline src/app/[locale]/main-page/page.tsx:8]: wrong imports priority - imports should come in a specific order: node modules, absolute path, relative path

File: src/app/[locale]/main-page/page.tsx [hunk 5]
```diff
@@ -1,10 +1,11 @@
...
 
-export default function MainPage() {
+function MainPage() {
   return (
```
Comment (AKutusenko) [inline src/app/[locale]/main-page/page.tsx:11]: no function return type

File: src/components/custom/propertiesList/index.tsx [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+  TextMiddleStyled,
+  CardMediaStyled,
+} from './styles';
+import Iconify from 'components/iconify';
+import { enqueueSnackbar } from 'notistack';
+import { QOBRIX_HOST } from 'config-global';
+import { fCurrency } from 'utils/format-number';
+import { AxiosError } from 'axios';
+import { IPropertyItem } from 'types/property';
+
```
Comment (AKutusenko) [inline src/components/custom/propertiesList/index.tsx:33]: wrong imports priority

File: src/components/custom/propertiesList/index.tsx [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+      })();
+  }, [isFetching, properties, propertiesError, propertiesPagination]);
+
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll);
+    return () => window.removeEventListener('scroll', handleScroll);
+  }, [propertiesPagination]);
+
+  function handleScroll() {
+    if (!propertiesPagination) {
```
Comment (AKutusenko) [inline src/components/custom/propertiesList/index.tsx]: no function return type

File: src/components/custom/propertiesList/index.tsx [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+    const docHeight = Math.max(
+      body.scrollHeight,
+      body.offsetHeight,
+      html.clientHeight,
+      html.scrollHeight,
+      html.offsetHeight
+    );
+    const windowBottom = windowHeight + window.scrollY;
+
+    if (windowBottom >= docHeight - 200 && propertiesPagination.hasNextPage) {
```
Comment (AKutusenko) [inline src/components/custom/propertiesList/index.tsx]: rule no magic numbers

File: src/api/property.ts (hunk 1 above)
Comment (serhii-rudenko) [reply src/api/property.ts]: fixed

File: src/utils/axios.ts (hunk 3 above)
Comment (serhii-rudenko) [reply src/utils/axios.ts]: fixed

File: src/types/property.ts (hunk 2 above)
Comment (serhii-rudenko) [reply src/types/property.ts]: fixed

File: src/components/custom/propertiesList/index.tsx (hunk 6 above)
Comment (serhii-rudenko) [reply src/components/custom/propertiesList/index.tsx:33]: fixed

File: src/components/custom/propertiesList/index.tsx (hunk 7 above)
Comment (serhii-rudenko) [reply src/components/custom/propertiesList/index.tsx]: fixed

File: src/components/custom/propertiesList/index.tsx (hunk 8 above)
Comment (serhii-rudenko) [reply src/components/custom/propertiesList/index.tsx]: fixed

**Review by AKutusenko — APPROVED** [review]

**Comment by AKutusenko** [comment]
проверь линтом  либо сделай билд проекта, почему проверка деплоя не прошла?

**Comment by rudenkoserhii** [comment]
> проверь линтом либо сделай билд проекта, почему проверка деплоя не прошла?

Перша помилка в файлі api/blog.ts який я взагалі не чипав. Шукаю...

File: src/components/custom/propertiesList/index.tsx [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+      html.scrollHeight,
+      html.offsetHeight
+    );
+    const windowBottom = windowHeight + window.scrollY;
+
+    if (windowBottom >= docHeight - FETCH_NEXT_BEFORE && propertiesPagination.hasNextPage) {
+      setIsFetching(true);
+    }
+  }
+
```
Comment (lyudad) [inline src/components/custom/propertiesList/index.tsx]: too many logic in component, move to custom hook

Comment (rudenkoserhii) [reply src/components/custom/propertiesList/index.tsx]: done

File: src/components/custom/propertiesList/index.tsx (hunk 6 above)
Comment (lyudad) [reply src/components/custom/propertiesList/index.tsx:33]: add to rules, it should be automatecally formatted by this time

File: src/components/custom/propertiesList/index.tsx (hunk 9 above)
Comment (lyudad) [reply src/components/custom/propertiesList/index.tsx]: https://www.notion.so/Git-rules-849f8ca8b3db4ded926d762c6fff8798 1.1.10 
ты смотрел чеклист? там 25 пунктов, это не так уж и много, за пару часов можно выучить

Comment (rudenkoserhii) [reply src/components/custom/propertiesList/index.tsx]: I thought eslint helping me always, but folder where I worked was in eslintignore. Now fixed

File: src/components/custom/propertiesList/index.tsx (hunk 6 above)
Comment (rudenkoserhii) [reply src/components/custom/propertiesList/index.tsx:33]: Done

File: src/app/[locale]/main-page/page.tsx (hunk 4 above)
Comment (lyudad) [reply src/app/[locale]/main-page/page.tsx:8]: add it to eslint rules. we feed fix it once and for forever

**Review by lyudad — APPROVED** [review]

**Pull request merged**